assembly/assemblyCommon/AssemblyAssemblyV2.cc

```diff
@@ -1268,6 +1270,128 @@ void AssemblyAssemblyV2::ApplyPSPToPSSXYOffset_finish()
+// ----------------------------------------------------------------------------------------------------
+void AssemblyAssemblyV2::MakeSpaceOnPlatform_start()
+{
+  if(in_action_){
+
+    NQLog("AssemblyAssemblyV2", NQLog::Warning) << "MakeSpaceOnPlatform_start"
+       << ": logic error, an assembly step is still in progress, will not take further action";
+
+    return;
+  }
+
+  const double dx0 = 0.0;
+  const double dy0 = 0.0;
+  const double dz0 = makespace_Z;
+  const double da0 = 0.0;
+
+  if(dz0 <= 0.)
+  {
+    NQLog("AssemblyAssemblyV2", NQLog::Critical) << "MakeSpaceOnPlatform_start"
+       << ": invalid (non-positive) value for vertical upward movement for makespace (dz=" << dz0 << "), no action taken";
+
+    NQLog("AssemblyAssemblyV2", NQLog::Spam) << "MakeSpaceOnPlatform_start"
+       << ": emitting signal \"MakeSpaceOnPlatform_finished\"";
+
+    emit MakeSpaceOnPlatform_finished();
+
+    return;
+  }
+
+  connect(this, SIGNAL(move_relative_request(double, double, double, double)), motion_, SLOT(moveRelative(double, double, double, double)));
+  connect(motion_, SIGNAL(motion_finished()), this, SLOT(MakeSpaceOnPlatform_finish()));
+
+  in_action_ = true;
+
+  NQLog("AssemblyAssemblyV2", NQLog::Spam) << "MakeSpaceOnPlatform_start"
+     << ": emitting signal \"move_relative_request(" << dx0 << ", " << dy0 << ", " << dz0 << ", " << da0 << ")\"";
+
+  emit move_relative_request(dx0, dy0, dz0, da0);
+}
+
+void AssemblyAssemblyV2::MakeSpaceOnPlatform_finish()
+{
+  disconnect(this, SIGNAL(move_relative_request(double, double, double, double)), motion_, SLOT(moveRelative(double, double, double, double)));
+  disconnect(motion_, SIGNAL(motion_finished()), this, SLOT(MakeSpaceOnPlatform_finish()));
+
+  if(in_action_){ in_action_ = false; }
+
+  NQLog("AssemblyAssemblyV2", NQLog::Spam) << "MakeSpaceOnPlatform_finish"
+     << ": emitting signal \"LiftUpPickupTool_finished\"";
+
+  emit MakeSpaceOnPlatform_finished();
+
+  NQLog("AssemblyAssemblyV2", NQLog::Message) << "MakeSpaceOnPlatform_finish"
+     << ": assembly-step completed";
+
+  emit DBLogMessage("== Assembly step completed : [Make space on platform]");
+}
+// ----------------------------------------------------------------------------------------------------
+
+// ----------------------------------------------------------------------------------------------------
+// Return to initial position -----------------------------------------------------------------------------------
+// ----------------------------------------------------------------------------------------------------
+void AssemblyAssemblyV2::ReturnToPlatform_start()
+{
+  if(in_action_){
+
+    NQLog("AssemblyAssemblyV2", NQLog::Warning) << "ReturnToPlatform_start"
+       << ": logic error, an assembly step is still in progress, will not take further action";
+
+    return;
+  }
+
+  const double dx0 = 0.0;
+  const double dy0 = 0.0;
+  const double dz0 = -makespace_Z;
```

Comment:
Seems slightly dangerous if someone moved the stage downwards manually in between steps. Should not happen, I know, but storing the old absolute z-position somewhere feels like the safer option. Not sure how easily that's feasible

Reply:
It now simply goes back to the old position - thanks for the hint :-)